components/match2/wikis/leagueoflegends/match_group_input_custom.lua

```diff
@@ -155,425 +147,201 @@ end
 
 CustomMatchGroupInput.processPlayer = FnUtil.identity
 
----@param data table
----@param indexedScores table[]
----@return table
----@return table[]
-function CustomMatchGroupInput.getResultTypeAndWinner(data, indexedScores)
-	-- Map or Match wasn't played, set not played
-	if
-		Table.includes(NP_STATUSES, data.finished) or
-		Table.includes(NP_STATUSES, data.winner)
-	then
-		data.resulttype = 'np'
-		data.finished = true
-	-- Map or Match is marked as finished.
-	-- Calculate and set winner, resulttype, placements and walkover (if applicable for the outcome)
-	elseif Logic.readBool(data.finished) then
-		if MatchGroupInput.isDraw(indexedScores) then
-			data.winner = 0
-			data.resulttype = 'draw'
-			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, 'draw')
-		elseif CustomMatchGroupInput.placementCheckSpecialStatus(indexedScores) then
-			data.winner = MatchGroupInput.getDefaultWinner(indexedScores)
-			data.resulttype = DEFAULT_RESULT_TYPE
-			if MatchGroupInput.hasForfeit(indexedScores) then
-				data.walkover = 'ff'
-			elseif MatchGroupInput.hasDisqualified(indexedScores) then
-				data.walkover = 'dq'
-			elseif MatchGroupInput.hasDefaultWinLoss(indexedScores) then
-				data.walkover = 'l'
-			end
-			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, DEFAULT_RESULT_TYPE)
-		else
-			local winner
-			indexedScores, winner = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, nil, data.finished)
-			data.winner = data.winner or winner
-		end
+---Should only be called on finished matches or maps
+---@param winner integer|string
+---@param opponents {score: number, status: string}[]
+---@return string? #Result Type
+---@return integer? #Winner
+---@return string? #Walkover
+function CustomMatchGroupInput.getResultTypeAndWinner(winner, opponents)
```

Comment:
Most wikis actually does not allow for the finished field on the match level, even if their code seems to allow for it. Simply because the finished field will be overwritten by the time it gets to this function

Reply:
imo it is a good improvement

Reply:
Yeah but almost no wiki where it looks like it's supported, it is acutally supported, which is not great